package/MDAnalysis/transformations/fit.py

```diff
@@ -0,0 +1,241 @@
+            ref_center = partial(reference.atoms.center_of_mass)
+            ag_center = partial(ag.atoms.center_of_mass)
+        else:
+            raise ValueError('{} is not a valid argument for "center_of"'.format(center_of))
+    except AttributeError:
+        if center_of == 'mass':
+            raise AttributeError('Either {} or {} is not an Universe/AtomGroup object with masses'.format(ag, reference))
+        else:
+            raise ValueError('Either {} or {} is not an Universe/AtomGroup object'.format(ag, reference))
+    
+    reference = np.asarray(ref_center(), np.float32)
+
+    def wrapped(ts):
+        center = np.asarray(ag_center(), np.float32)
+        vector = reference - center
+        if plane:
+            vector[plane] = 0
+        ts.positions += vector
+        
+        return ts
+    
+    return wrapped
+
+
+def fit_rot_trans(ag, reference, plane=None, weights=None):
+    """Perform a spatial superposition by minimizing the RMSD.
+
+    Spatially align the group of atoms `ag` to `reference` by doing a RMSD
+    fit on `select` atoms.
+    
+    This fit works as a way o remove translations and rotations of a given
+    AtomGroup in a trajectory. A plane can be given using the flag `plane`
+    so that only translations and rotations in that particular plane are
+    removed. This is useful for protein-membrane systems to where the membrane
+    must remain in the same orientation.
+    
+    Example
+    -------
+    Removing the translations and rotations of a given AtomGroup `ag` on the XY plane
+    by fitting it to a reference `ref`, using the masses as weights for the RMSD fit.
+    
+    ..code-block::python
+    
+        ag = u.select_atoms("protein")
+        ref = mda.Universe("reference.pdb")
+        transform = MDAnalysis.transformations.fit_rot_trans(ag, ref, plane="xy", weights="mass")
+        u.trajectory.add_transformations(transform)
+    
+    Parameter
+    ---------
+    ag : Universe or AtomGroup
+       structure to translate, a
+       :class:`~MDAnalysis.core.groups.AtomGroup` or a whole 
+       :class:`~MDAnalysis.core.universe.Universe`
+    reference : Universe or AtomGroup
+       reference structure, a :class:`~MDAnalysis.core.groups.AtomGroup` or a whole 
+       :class:`~MDAnalysis.core.universe.Universe`
+    plane: str, optional
+        used to define the plane on which the translations will be removed. Defined as a 
+        string of the plane. Suported planes are "yz", "xz" and "xy" planes.
+    weights : {"mass", ``None``} or array_like, optional
+       choose weights. With ``"mass"`` uses masses as weights; with ``None``
+       weigh each atom equally. If a float array of the same length as
+       `ag` is provided, use each element of the `array_like` as a
+       weight for the corresponding atom in `mobile`.
+       
+    Returns
+    -------
+    MDAnalysis.coordinates.base.Timestep
+    """
+    if plane is not None:
+        if plane not in ('xy', 'yz', 'xz'):
```

Comment:
If you define `axes` first, you can do `if plane not in axes`. You can also do:

```
try:
    plane = axes[plane]
except KeyError:
    raise ValueError('{} is not a valid plane'.format(plane))
```

Reply:
looks like this has been done